File: leetcode/LinkedListTests/LinkedList_203.cs
namespace LinkedListTests;

[TestFixture]
internal class LinkedList_203
{
    private class Solution {
        public ListNode RemoveElements(ListNode head, int val) {
            var dummyHead = new ListNode();
            var prev = dummyHead;
            var curr = head;
            prev.next = curr;
            while(curr is not null) {
                //prev->curr->next
                //prev->next
                if(curr.val == val) {
                    prev.next = curr.next;
                } else {
                    prev = prev.next;
                }
                curr = curr.next;            
            }
            return dummyHead.next;
        }
    }
}

File: leetcodePython/LinkedList/linkedlist_203.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummyHead = ListNode()
        dummyHead.next = head
        prev = dummyHead
        curr = head
        while curr:
            if curr.val == val:
                prev.next = curr.next
            else:
                prev = curr
            curr = curr.next

        return dummyHead.next
